litestar/utils/helpers.py

```diff
@@ -70,3 +83,51 @@ def unwrap_partial(value: MaybePartial[T]) -> T:
     while hasattr(output, "func"):
         output = output.func
     return cast("T", output)
+
+
+def encode_headers(
+    headers: Iterable[tuple[str, Any]], cookies: Iterable[Cookie], raw_headers: list[tuple[bytes, bytes]]
+) -> list[tuple[bytes, bytes]]:
+    """Encode the response headers as a list of byte tuples.
+
+    Args:
+        headers: Iterable of header name/value pairs.
+        cookies: A list of cookies.
+        raw_headers: A list of raw headers.
+
+    Returns:
+        A list of byte tuples.
+    """
+    return list(
+        chain(
+            ((k.lower().encode("latin-1"), str(v).encode("latin-1")) for k, v in headers),
+            (cookie.to_encoded_header() for cookie in cookies if not cookie.documentation_only),
+            raw_headers,
+        )
+    )
+
+
+def filter_cookies(local_cookies: Iterable[Cookie], layered_cookies: Iterable[Cookie]) -> list[Cookie]:
+    """Given two sets of cookies, return a unique list of cookies, that are not marked as documentation_only.
+
+    Args:
+        local_cookies: Cookies returned from the local scope.
+        layered_cookies: Cookies returned from the layers.
+
+    Returns:
+        A unified list of cookies
+    """
+    # TODO: check this precedence, should local cookies be overwritten by layered cookies?
```

Comment:
Should be the other way around, right?

Reply:
There's actually no issue here - I just wrote the TODO to ensure I circled back to take a look, but the local headers unpacked into the set first means they take precedence. I see way more of this pattern to unpack into mappings where the 2nd unpacked takes precedence, and was looking at this with that frame of mind.